### PR TITLE
868 improving README with AWS API gateway findings

### PR DESCRIPTION
**Pull request by fahernandez**

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Hi Fernando! Sorry this hasn't had a review yet, I had an idea! How would you feel about adding this to the docs pages instead of the readme m

**Comment by fahernandez** [comment]
Sure, no problem at all! But what you mean with docs pages? Can you share the file path where you consider is a better idea to put it?

**Comment by fahernandez** [comment]
> Hi Fernando! Sorry this hasn't had a review yet, I had an idea! How would you feel about adding this to the docs pages instead of the readme?

Sure, no problem at all! But what you mean with docs pages? Can you share the file path where you consider is a better idea to put it?

**Comment by johanbrandhorst** [comment]
Look at `docs/_docs/patch.md` for example. We could add an `aws.md` which has your tips in it, and we could link to it from `index.md`. What do you think?

**Comment by fahernandez** [comment]
> Look at `docs/_docs/patch.md` for example. We could add an `aws.md` which has your tips in it, and we could link to it from `index.md`. What do you think?

It seems good to me. Please look at the changes and let me know if it's ok.

File: docs/index.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ To learn more about us check out our documentation on:
 *   [Installation and usage](_docs/usage.md)
 *   [Examples](_docs/examples.md)
 *   [Features](_docs/features.md)
+*   [Aws](_docs/aws.md)
```
Comment (johanbrandhorst) [inline docs/index.md]:
```suggestion
*   [AWS API Gateway tips](_docs/aws.md)
```

File: docs/_docs/aws.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+category: documentation
+---
+
+# AWS
+
+## Import swagger documentation into AWS API Gateway
+AWS Api gateway service allows to import swagger documentation to create REST API. The process is very straigh foward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
The AWS API gateway service allows importing of a OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```

File: docs/_docs/aws.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+These are some considerations when importing the documentation:
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
Here are some tips to consider when importing the documentation:
```

File: docs/_docs/aws.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+1. Remove model circular dependencies(Circular dependencies aren't supported by the parser).
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
1. Remove any circular dependencies (these aren't supported by the parser).
```

File: docs/_docs/aws.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+2. Remove security-related annotations (These annotations aren't well supported by the parser).
+3. Max length of fields is reviewed by the parser but the errors aren't self-explanatory. Review the [documentation](https://swagger.io/specification/v2/) to verify that the requirements are met.
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
3. Max length of fields are reviewed by the parser but the errors aren't self-explanatory. Review the [documentation](https://swagger.io/specification/v2/) to verify that the requirements are met.
```

File: docs/_docs/aws.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+4. API gateway errors aren't good enough for troubleshooting so I recommend using this [page](https://apidevtools.org/swagger-parser/online/) for structure validation.
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
4. API gateway errors aren't great, but you can use this [page](https://apidevtools.org/swagger-parser/online/) for structure validation.
```

**Comment by johanbrandhorst** [comment]
Just a few grammatical edits, looks good!

File: docs/index.md (hunk 1 above)
Comment (fahernandez) [reply docs/index.md]: ok

File: docs/_docs/aws.md (hunk 2 above)
Comment (fahernandez) [reply docs/_docs/aws.md]: ok

File: docs/_docs/aws.md (hunk 3 above)
Comment (fahernandez) [reply docs/_docs/aws.md]: ok

File: docs/_docs/aws.md (hunk 4 above)
Comment (fahernandez) [reply docs/_docs/aws.md]: ok

File: docs/_docs/aws.md (hunk 5 above)
Comment (fahernandez) [reply docs/_docs/aws.md]: ok

**Comment by fahernandez** [comment]
Typos were fixed as Johan suggested

File: docs/_docs/aws.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
+category: documentation
+---
+
+# AWS
+
+## Import swagger documentation into AWS API Gateway
+The AWS API gateway service allows importing of a OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```
Comment (johanbrandhorst) [inline docs/_docs/aws.md]:
```suggestion
The AWS API gateway service allows importing of an OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```

**Comment by johanbrandhorst** [comment]
Just discovered a small typo in my own suggestion 😅. Thanks again for this.

**Comment by johanbrandhorst** [comment]
Ah curses, accepting my own suggesting messes with the CLA. Uhm, let me get back to you on what we can do to fix this.

**Comment by fahernandez** [comment]
> Ah curses, accepting my own suggesting messes with the CLA. Uhm, let me get back to you on what we can do to fix this.

Sorry for that, I thought it was faster only accepting the suggested change.

**Comment by johanbrandhorst** [comment]
Not your fault, we'll figure this out on our end and get this merged or let you know what needs to be done :).

**Comment by johanbrandhorst** [comment]
I consent to merging my contribution of this pull request.

**Pull request merged**